src/widget/wcoverartmenu.cpp

```diff
@@ -82,8 +88,28 @@ void WCoverArtMenu::slotChange() {
     coverInfo.source = CoverInfo::USER_SELECTED;
     coverInfo.coverLocation = selectedCoverPath;
     coverInfo.setImage(image);
-    qDebug() << "WCoverArtMenu::slotChange emit" << coverInfo;
-    emit coverInfoSelected(coverInfo);
+
+    QString coverArtCopyFilePath =
+            trackFileInfo.absoluteFilePath().left(
+                    trackFileInfo.absoluteFilePath().lastIndexOf('.') + 1) +
+            selectedCoverExtension;
+
+    if (QFileInfo(m_coverInfo.trackLocation).canonicalPath() ==
+            QFileInfo(selectedCoverPath).canonicalPath()) {
+        qDebug() << "Track and selected cover art are in the same path:"
+                 << QFileInfo(selectedCoverPath).canonicalPath()
+                 << "Cover art updated without copying";
+        emit coverInfoSelected(coverInfo);
+        qDebug() << "WCoverArtMenu::slotChange emit" << coverInfo;
+        return;
+    }
+
+    m_worker.reset(new CoverArtCopyWorker(image, coverArtCopyFilePath));
```

Comment:
This smart pointer will delete the worker even though it might still be running. 
We need to wait for the task to be finished in the destructor of this class. Check out other places in the code for a template.

Reply:
I've added a quit() in the last method of the worker, and this triggers a bool member in WCoverArtMenu, I've added a VERIFY_OR_DEBUG_ASSERT(m_isWorkerRunning == false), IDK if that solves this issue.